Radzen.Blazor/RadzenDatePicker.razor.cs

```diff
@@ -547,6 +547,10 @@ protected async Task ParseDate()
                     DateTimeOffset? offset = DateTime.SpecifyKind((DateTime)Value, Kind);
                     await ValueChanged.InvokeAsync((TValue)(object)offset);
                 }
+                else if (Value != null)
+                {
+                    await ValueChanged.InvokeAsync((TValue)(object)DateTime.SpecifyKind((DateTime)Value, Kind));
```

Comment:
Well, there is a problem with DatePicker. Kind value is lost when updating the value and is defaulted to unspecified. I'm not too confident about this casting `(DateTime)Value` so please advise.

Reply:
added also type checking